lib/huginn_record_link_agent/record_link_agent.rb

```diff
@@ -600,75 +537,99 @@ def build_events(event, data, lookup_record, source_records = nil, target_record
         # Emit each event individually
 
         sources.each do |s|
-          payload = base_event.merge({
-            record_link: {
-              link_status: "200",
-              source_system: s[:system],
-              source_type: s[:type],
-              source_id: s[:uid],
-              target_system: lookup_record.ext_system,
-              target_type: lookup_record.model_type,
-              target_id: lookup_record.uid
-            }
-          })
 
-          create_event payload: payload
+          record_link = create_record_link_payload(
+            source_system: s[:system],
+            source_type: s[:type],
+            source_id: s[:external_id],
+            target_system: lookup_record.ext_system,
+            target_type: lookup_record.model_type,
+            target_id: lookup_record.external_id
+          )
+
+          emit(data, event, { record_link: record_link }) if boolify(options['emit_events'])
         end
 
         targets.each do |t|
-          payload = base_event.merge({
-            record_link: {
-              link_status: "200",
-              source_system: lookup_record.ext_system,
-              source_type: lookup_record.model_type,
-              source_id: lookup_record.uid,
-              target_system: t[:system],
-              target_type: t[:type],
-              target_id: t[:uid]
-            }
-          })
 
-          create_event payload: payload
+          record_link = create_record_link_payload(
```

Comment:
@mrogers-5s same thing for `create_record_link_payload`. Use parameters as it better enforces the interface.

Reply:
Consider it done. I opted not to do that initially given that there were slight variations in the structure of the output based on what was being requested.

I was concerned consolidating would make the flow of event creation harder to follow in the future. I think I found a decent way to pull it off